expertsystem/ui/_system_control.py

```diff
@@ -170,9 +217,9 @@ def remove_duplicate_solutions(
     return filtered_results
 
 
-def _remove_qns_from_graph(
-    graph, qn_list
-):  # pylint: disable=too-many-branches
+def _remove_qns_from_graph(  # pylint: disable=too-many-branches
+    graph: StateTransitionGraph, qn_list: list
```

Comment:
If you want it to be precise qn_list should also be a `Union[InteractionQuantumNumberNames, StateQuantumNumberNames, ParticlePropertyNames]`

Reply:
Could be that I tried, but that things blew up elsewhere. So I practically stated with `List[Any]` here.

Reply:
Ok, if it doesnt work just leave it like this. Having the types pinned down more than Any is already useful. Through the refactoring everything will change anyways, so we do not need to be super critical here.

Reply:
Ok works actually 2019d1810c4cfc12db6e9ffa7d1d27fb3cd8a115